### ISSUE TITLE
Jinja Template Support for Stored GraphQL Queries

### ISSUE THREAD

**Issue by jifox** [nautobot/nautobot#6085]
### As ...

Nelly - Network Engineer

### I want ...

 to specify Jinja templates for stored GraphQL queries in Nautobot so that I can optionally render the GraphQL output. This will make it easier for me and other users to read and use the query results, such as producing hyperlinks to access selected objects

### So that ...

it easier for users to read and utilize query results, enabling the production of hyperlinks to access selected objects and more.

**Benefits:**

- **Improved Readability**: Users can format the output to be more understandable and visually appealing.
- **Enhanced Usability**: By rendering the output with Jinja templates, users can include hyperlinks and other interactive elements, making it easier to navigate and utilize the data.
- **Customization**: Users can tailor the output to meet their specific needs and preferences, enhancing the overall user experience

### I know this is done when...

- I create GraphQL queries and add/edit a jinja template that will have the query reaults available for rendering
- I activate the checkbox [ ] Use the template to format the query result

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Apps (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

- New Jinja template for GraphiQL queries

**Comment by itdependsnetworks** [comment]
I think this is the same as https://github.com/nautobot/nautobot-app-golden-config/issues/681

**Comment by jifox** [comment]
@itdependsnetworks Yes, it is nearly the same, but I don’t want it to be limited to devices. I envision that any GraphQL query will be accepted.
